rasa_core/domain.py

```diff
@@ -575,9 +580,19 @@ def persist_clean(self, filename):
                 if intent.get("use_entities"):
                     data["intents"][idx] = name
 
-        for name, slot in data["slots"].items():
+        for slot in data["slots"].values():
             if slot["initial_value"] is None:
                 del slot["initial_value"]
+            if slot["auto_fill"] is True:
+                del slot["auto_fill"]
+            if slot["type"].startswith('rasa_core.slots'):
+                slot["type"] = Slot.resolve_by_type(slot["type"]).type_name
+
+        if data["config"]["store_entities_as_slots"] is True:
+            del data["config"]["store_entities_as_slots"]
+
+        # clean empty keys
+        data = {k: v for k, v in data.items() if v or v is False}
```

Comment:
what do you want to filter out here? empty lists / dicts? If so, please make that explicit
```suggestion
        data = {k: v for k, v in data.items() if v == {} or v == [] or v is None}
```